drivers/adc/adc_ti_k3.c

```diff
@@ -0,0 +1,389 @@
+/*
+ * Copyright (c) 2025 Texas Instruments Incorporated
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_adc_k3
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+#include <zephyr/drivers/adc.h>
+
+#define ADC_CONTEXT_USES_KERNEL_TIMER
+#include "adc_context.h"
+
+#include <zephyr/logging/log.h>
+#include <zephyr/irq.h>
+LOG_MODULE_REGISTER(adc_k3);
+
+#define ADC_K3_TOTAL_CHANNELS 8
+#define ADC_K3_TOTAL_STEPS    16
+
+struct adc_k3_regs {
+	uint8_t RESERVED_1[0x28]; /**< Reserved, offset: 0xA0 - 0x28 */
```

Comment:
@natto1784 can you please replace all `__IO` with `volatile`. These `_IO` definitions are not c-standard and coming from the STM32 HAL.

This should also fix these `checkpatch.pl` warnings.

Reply:
It actually comes from the cmsis module but it is an HAL regardless so I think it is better to do as you say.

Reply:
I will update `drivers/i2c/i2c_omap.c`. I saw there is an empty line in the struct. Probably to get rid of this warning.